spec/models/miq_queue_spec.rb

```diff
@@ -595,4 +595,13 @@ def self.some_method(single_arg)
                                   :expanded     => [nil, "exp"]
                                 )
   end
+
+  context "#delivered" do
+    it "destroys a stale object" do
+      q = MiqQueue.create!(:state => 'ready')
+      MiqQueue.find(q.id).tap { |q2| q2.state = 'dequeue' }.save # update_attributes doesn't expose the issue
+
+      expect { q.delivered('warn', nil, nil) }.not_to raise_exception
```

Comment:
Can we do this instead?

```ruby
q.delivered('warn', nil, nil)
expect { q.reload }.to raise_exception(ActiveRecord::RecordNotFound)
```

Either that or that there isn't a MiqQueue with the q.id.